### DISCUSSION TITLE
graphscope复现ldbc测试query7有一半左右错误，请问原因

### DISCUSSION THREAD

**Discussion by timer-2117** [alibaba/GraphScope#3122] [Q&A] [answered]
如题，使用ldbc官网的附件，生成10g数据跑正确性测试，query7一半错误，和预期结果有差距
![image](https://github.com/alibaba/GraphScope/assets/53077051/36986196-6d32-4bc0-9944-55b62218ae41)
部分结果差异
![image](https://github.com/alibaba/GraphScope/assets/53077051/c0ab0058-639d-4358-b83e-9214896bd8dc)

config/validate.properties配置文件内容：
url=http://192.168.1.76:10000
readTimeout=5000000
connectTimeout=5000000
connectPoolMaxIdle=10
keepAliveDuration=5000
maxRequestsPerHost=180
maxRequests=180

printQueryNames=false
printQueryStrings=false
printQueryResults=false

status=1
thread_count=1
mode=validate_database
name=LDBC-SNB
results_log=false
time_unit=MILLISECONDS
time_compression_ratio=0.001
peer_identifiers=
workload_statistics=false
spinner_wait_duration=0
help=false
ignore_scheduled_start_times=true
vdb=/graphscope/validation_params-sf10.csv

ldbc.snb.interactive.update_interleave=895969
workload=org.ldbcouncil.snb.driver.workloads.interactive.LdbcSnbInteractiveWorkload
db=org.ldbcoucil.snb.impls.workloads.graphscope.interactive.GraphScopeInteractiveDb

operation_count=10000

ldbc.snb.interactive.updates_dir=/test_data_10/social_network
ldbc.snb.interactive.parameters_dir=/test_data_10/substitution_parameters/
ldbc.snb.interactive.short_read_dissipation=0.2
ldbc.snb.interactive.scale_factor=10

ldbc.snb.interactive.LdbcQuery1_freq=1
ldbc.snb.interactive.LdbcQuery2_freq=1
ldbc.snb.interactive.LdbcQuery3_freq=1
ldbc.snb.interactive.LdbcQuery4_freq=1
ldbc.snb.interactive.LdbcQuery5_freq=1
ldbc.snb.interactive.LdbcQuery6_freq=1
ldbc.snb.interactive.LdbcQuery7_freq=1
ldbc.snb.interactive.LdbcQuery8_freq=1
ldbc.snb.interactive.LdbcQuery9_freq=1
ldbc.snb.interactive.LdbcQuery10_freq=1
ldbc.snb.interactive.LdbcQuery11_freq=1
ldbc.snb.interactive.LdbcQuery12_freq=1
ldbc.snb.interactive.LdbcQuery13_freq=1
ldbc.snb.interactive.LdbcQuery14_freq=1

ldbc.snb.interactive.LdbcQuery1_enable=true
ldbc.snb.interactive.LdbcQuery2_enable=true
ldbc.snb.interactive.LdbcQuery3_enable=true
ldbc.snb.interactive.LdbcQuery4_enable=true
ldbc.snb.interactive.LdbcQuery5_enable=true
ldbc.snb.interactive.LdbcQuery6_enable=true
ldbc.snb.interactive.LdbcQuery7_enable=true
ldbc.snb.interactive.LdbcQuery8_enable=true
ldbc.snb.interactive.LdbcQuery9_enable=true
ldbc.snb.interactive.LdbcQuery10_enable=true
ldbc.snb.interactive.LdbcQuery11_enable=true
ldbc.snb.interactive.LdbcQuery12_enable=true
ldbc.snb.interactive.LdbcQuery13_enable=true
ldbc.snb.interactive.LdbcQuery14_enable=true

ldbc.snb.interactive.LdbcShortQuery1PersonProfile_enable=true
ldbc.snb.interactive.LdbcShortQuery2PersonPosts_enable=true
ldbc.snb.interactive.LdbcShortQuery3PersonFriends_enable=true
ldbc.snb.interactive.LdbcShortQuery4MessageContent_enable=true
ldbc.snb.interactive.LdbcShortQuery5MessageCreator_enable=true
ldbc.snb.interactive.LdbcShortQuery6MessageForum_enable=true
ldbc.snb.interactive.LdbcShortQuery7MessageReplies_enable=true

ldbc.snb.interactive.LdbcUpdate1AddPerson_enable=true
ldbc.snb.interactive.LdbcUpdate2AddPostLike_enable=true
ldbc.snb.interactive.LdbcUpdate3AddCommentLike_enable=true
ldbc.snb.interactive.LdbcUpdate4AddForum_enable=true
ldbc.snb.interactive.LdbcUpdate5AddForumMembership_enable=true
ldbc.snb.interactive.LdbcUpdate6AddPost_enable=true
ldbc.snb.interactive.LdbcUpdate7AddComment_enable=true
ldbc.snb.interactive.LdbcUpdate8AddFriendship_enable=true

**Comment by luoxiaojian** [comment, marked as answer]
这个可能是闰秒的问题，我们audit使用的是v0.3.6的specification和对应的driver，最新版本的specification里IC7的定义里有一些对于闰秒的修正：

Validation rule: Depending on whether the system-under-test supports leap seconds or uses UTC-SLS (UTC with Smoothed Leap Seconds), a difference of 1 minute can occur between the minutesLatency results of two correct implementations when the time interval includes June 30, 2012, when there was a leap second. Therefore, the minutesLatency value is validated using a tolerance of 1 minute.

如果照这个描述，看起来这些diff都在1分钟以内，是能够接受的（但那个版本的driver还是判错了）

能否麻烦你提供一下你的10g数据及正确结果的生成方式呢？谢谢
我们想尝试复现一下这些报错，针对错误日志，也可以确定是否为闰秒的问题

**Reply by timer-2117** [reply]
感谢回复~10g数据是通过datagen生成的，使用方式参考这个帖子https://blog.csdn.net/spquan/article/details/121455546，正确的结果是在这个测试报告中第六页脚注部分下载的压缩包https://ldbcouncil.org/benchmarks/snb/LDBC_SNB_I_20230713_SF30-100-300_GraphScopeFlex.pdf

**Reply by luoxiaojian** [reply]
我们使用最新版本的datagen（v0.3.8）做了测试，复现了这个问题

经排查，原因是ldbc提供的validation_params.csv是从v0.3.6版本的datagen生成的图生成的（如果你使用v0.3.6版本的datagen生成的图，应该是能通过正确性验证的）

这个联系了ldbc的工作人员，他们会基于v0.3.8的datagen重新生成一组正确结果，并近几天就会上传上去
